### PR TITLE
feat: support sound and title in `TechulusPush`

### PR DESCRIPTION
**Pull request by arjunkomath**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Update Push by Techulus integration to include more features, I'm the founder and creator of Push by Techulus.
https://github.com/techulus

## Type of change

- User interface (UI)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots

| Event | Before | After |
|-------|--------|-------|
| UI | - | CleanShot 2024-10-09 at 23 54 35@2x |
| Pressing the Testing Button in the app | CleanShot 2024-10-09 | IMG_6B137624A276-1 |

### CODE REVIEW

File: src/components/notifications/TechulusPush.vue [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+        <input id="push-api-title" v-model="$parent.notification.pushTitle" type="text" class="form-control">
+    </div>
+
+    <div class="mb-3">
+        <label for="push-api-channel" class="form-label">{{ $t("Channel") }}</label>
+        <input id="push-api-channel" v-model="$parent.notification.pushChannel" type="text" class="form-control">
+    </div>
+
+    <div class="mb-3">
+        <label for="push-api-sound" class="form-label">{{ $t("Sound") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/TechulusPush.vue:18]: Please make sure that you have added every translation string to [`en.json`](https://github.com/louislam/uptime-kuma/blob/master/src/lang/en.json). If they are not, they won't be avalibale for translation from our awesome group of translators.

File: server/notification-providers/techulus-push.js [hunk 2]
```diff
@@ -10,11 +10,22 @@ class TechulusPush extends NotificationProvider {
     async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
         const okMsg = "Sent Successfully.";
 
+        let data = {
+            "title": notification?.pushTitle?.length ? notification.pushTitle : "Uptime-Kuma",
```
Comment (CommanderStorm) [inline server/notification-providers/techulus-push.js:14]: Please add this default to the frontend as well.

Here is how this works

https://github.com/louislam/uptime-kuma/blob/e8c650797c80fe7189091afa9a81d4bad679dac1/src/components/notifications/Ntfy.vue#L66-L80

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Mostly looks good.
I have made some comments about the UI part.

For testing notification porovider we require at least one screenshot of the notification provider working, could you please update the part of the description which I edited? 😉

File: server/notification-providers/techulus-push.js (hunk 2 above)
Comment (arjunkomath) [reply server/notification-providers/techulus-push.js:14]: Thanks, I've added this.

File: src/components/notifications/TechulusPush.vue [hunk 3]
```diff
@@ -4,6 +4,53 @@
...
     </div>
 
+    <div class="mb-3">
+        <label for="push-api-title" class="form-label">{{ $t("Title") }}</label>
+        <input id="push-api-title" v-model="$parent.notification.pushTitle" type="text" class="form-control">
+    </div>
+
+    <div class="mb-3">
+        <label for="push-api-channel" class="form-label">{{ $t("Channel") }}</label>
+        <input id="push-api-channel" v-model="$parent.notification.pushChannel" type="text" class="form-control">
```
Comment (CommanderStorm) [inline src/components/notifications/TechulusPush.vue:14]: lets add a `pattern` here to make the helptext enforcement simpler

```suggestion
        <input id="push-api-channel" v-model="$parent.notification.pushChannel" type="text" class="form-control" patttern="[A-Za-z0-9-]+">
```

File: src/components/notifications/TechulusPush.vue [hunk 4]
```diff
@@ -4,6 +4,53 @@
...
+            <option value="money">{{ $t("Money") }}</option>
+            <option value="scifi">{{ $t("Scifi") }}</option>
+            <option value="clear">{{ $t("Clear") }}</option>
+            <option value="elevator">{{ $t("Elevator") }}</option>
+            <option value="guitar">{{ $t("Guitar") }}</option>
+            <option value="pop">{{ $t("Pop") }}</option>
+        </select>
+        <div class="form-text">
+            {{ $t("Custom Notification sound") }}
+        </div>
```
Comment (CommanderStorm) [inline src/components/notifications/TechulusPush.vue:41]: I think the helptext would be a better title

```suggestion
        <label for="push-api-sound" class="form-label">{{ $t("Custom Notification sound") }}</label>
        <select id="push-api-sound" v-model="$parent.notification.pushSound" class="form-select">
            <option value="default">{{ $t("Default") }}</option>
            <option value="arcade">{{ $t("Arcade") }}</option>
            <option value="correct">{{ $t("Correct") }}</option>
            <option value="fail">{{ $t("Fail") }}</option>
            <option value="harp">{{ $t("Harp") }}</option>
            <option value="reveal">{{ $t("Reveal") }}</option>
            <option value="bubble">{{ $t("Bubble") }}</option>
            <option value="doorbell">{{ $t("Doorbell") }}</option>
            <option value="flute">{{ $t("Flute") }}</option>
            <option value="money">{{ $t("Money") }}</option>
            <option value="scifi">{{ $t("Scifi") }}</option>
            <option value="clear">{{ $t("Clear") }}</option>
            <option value="elevator">{{ $t("Elevator") }}</option>
            <option value="guitar">{{ $t("Guitar") }}</option>
            <option value="pop">{{ $t("Pop") }}</option>
        </select>
```

File: src/components/notifications/TechulusPush.vue [hunk 5]
```diff
@@ -4,6 +4,53 @@
         <HiddenInput id="push-api-key" v-model="$parent.notification.pushAPIKey" :required="true" autocomplete="new-password"></HiddenInput>
     </div>
 
+    <div class="mb-3">
+        <label for="push-api-title" class="form-label">{{ $t("Title") }}</label>
+        <input id="push-api-title" v-model="$parent.notification.pushTitle" type="text" class="form-control">
+    </div>
+
+    <div class="mb-3">
+        <label for="push-api-channel" class="form-label">{{ $t("Channel") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/TechulusPush.vue:13]:
```suggestion
        <label for="push-api-channel" class="form-label">{{ $t("Notification channel") }}</label>
```

File: src/components/notifications/TechulusPush.vue [hunk 6]
```diff
@@ -4,6 +4,53 @@
...
+        <input id="push-api-channel" v-model="$parent.notification.pushChannel" type="text" class="form-control">
+        <div class="form-text">
+            {{ $t("Notification channel, alphanumerical string and hyphens only.") }}
```
Comment (CommanderStorm) [inline src/components/notifications/TechulusPush.vue:16]:
```suggestion
            {{ $t("Alphanumerical string and hyphens only.") }}
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
A few things left.

Please adresss https://github.com/louislam/uptime-kuma/pull/5178#discussion_r1793395832 though.
Given that you have started the PR from `master`, I cannot do this myself.

**Comment by arjunkomath** [comment]
> A few things left.
> 
> Please adresss [#5178 (comment)](https://github.com/louislam/uptime-kuma/pull/5178#discussion_r1793395832) though. Given that you have started the PR from `master`, I cannot do this myself or apply suggestions..

sure, I'll fix all of these today

File: src/components/notifications/TechulusPush.vue (hunk 1 above)
Comment (arjunkomath) [reply src/components/notifications/TechulusPush.vue:21]: Done!

**Review by CommanderStorm — APPROVED** [review]
Thanks for adjustment to the notification provider! ❤️



> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>

**Pull request merged**